docs/hub/spaces-oauth.md

```diff
@@ -0,0 +1,77 @@
+# Adding a Sign-In with HF button to your Space
+
+You can enable a built-in sign-in flow in your Space by seamlessly creating and associating an [OAuth/OpenID connect](https://developer.okta.com/blog/2019/10/21/illustrated-guide-to-oauth-and-oidc) app so users can log in with their HF account.
+
+This enables new use cases for your Space. For instance, when combined with [Persistent Storage](https://huggingface.co/docs/hub/spaces-storage), a generative AI Space could allow users to log in to access their previous generations, only accessible to them.
+
+## Create an OAuth app
+
+All you need to do is add `hf_oauth: true` to your Space's metadata inside your `README.md` file.
+
+Here's an example of metadata for a Gradio Space:
+
+```yaml
+title: Gradio Oauth Test
+emoji: 🏆
+colorFrom: pink
+colorTo: pink
+sdk: gradio
+sdk_version: 3.36.1
+python_version: 3.10.6
+app_file: app.py
+
+hf_oauth: true
+hf_oauth_redirect_path: /custom_callback_route # optional, see "Redirect URLs" below
+```
+
+You can check out the [configuration reference docs](./spaces-config-reference) for more information.
+
+This will add the following [environment variables](https://huggingface.co/docs/hub/spaces-overview#helper-environment-variables) to your space:
+
+- `OAUTH_CLIENT_ID`: the client ID of your OAuth app (public)
+- `OAUTH_CLIENT_SECRET`: the client secret of your OAuth app
+- `OAUTH_SCOPES`: scopes accessible by your OAuth app. Currently, this is always `"openid profile"`.
+- `OPENID_PROVIDER_URL`: The URL of the OpenID provider. The OpenID metadata will be available at [`{OPENID_PROVIDER_URL}/.well-known/openid-configuration`](https://huggingface.co/.well-known/openid-configuration).
+
+As for any other environment variable, you can use them in your code by using `os.getenv("OAUTH_CLIENT_ID")`, for example.
+
+## Redirect URLs 
+
+The allowed redirect URIs for your OAuth app are:
+
+- `https://{SPACE_SUBDOMAIN}.hf.space/auth/callback`
+- `https://{SPACE_SUBDOMAIN}.hf.space/login/callback`
+
+Note that `SPACE_SUBDOMAIN` is also [available](https://huggingface.co/docs/hub/spaces-overview#helper-environment-variables) as an environment variable.
+
+You can add a custom relative redirect path by setting `hf_oauth_redirect_path` in your Space's metadata.
+
+## Scopes
+
+The following scopes are available:
+
+- `openid`: Get the ID token in addition to the access token.
+- `profile`: Get the user's profile information (username, avatar, etc.)
+
+You should use `"openid profile"` as the scope for your OAuth app.
+
+## Adding the button to your Space
+
+You now have all the information to add a "Sign-in with HF" button to your Space. Libraries in [various](https://github.com/lepture/authlib) [languages](https://github.com/panva/node-openid-client) can help you implement the OpenID/OAuth protocol. Gradio also provides **built-in support**, making implementing the Sign-in with HF button a breeze; you can [check out the associated doc](TODO).
```

Comment:
ping @Wauplin regarding the `TODO`, not sure there's a doc for Gradio support already or not? feel free to push changes directly!

Reply:
There's a small guide yes. See https://github.com/huggingface/hub-docs/commit/434c3b4fda7f54474ed2083fed36717b9210160e